### ISSUE TITLE
Secrets stored as symbolic link

### ISSUE THREAD

**Issue by g14como** [Azure/secrets-store-csi-driver-provider-azure#656]
**Have you**

- [x] Read [Troubleshooting Guide](https://azure.github.io/secrets-store-csi-driver-provider-azure/troubleshooting/)
- [x] Searched on [GitHub issues](https://github.com/Azure/secrets-store-csi-driver-provider-azure/issues) and [Discussions](https://github.com/Azure/secrets-store-csi-driver-provider-azure/discussions)

but didn't find anything related to the issue described below

**What steps did you take and what happened:**
Upgrading CSI driver from v0.0.22 to v0.3.0 and provider azure from 0.0.15 to v0.2.0

After the upgrade when i redeployed my applications i saw that secrets are stored in my mountPath as a symbolic link to another folder 

pwd: /mnt/secrets
drwxr-xr-x 2 root root 380 Sep 27 10:17 ..2021_09_27_08_17_18.3114856258/
lrwxrwxrwx 1 root root  32 Sep 27 10:17 ..data -> ..2021_09_27_08_17_18.3114856258/
lrwxrwxrwx 1 root root  40 Sep 27 10:17 endpoint-foo -> ..data/endpoint-foo

**What did you expect to happen:**

i expect to have secrets stored directly in my mountPath and without any symbolic link as below

pwd: /mnt/secrets
-rw-r--r-- 1 root root   6 Sep 27 10:17 endpoint-foo

**Which access mode did you use to access the Azure Key Vault instance:**
Service Principal

**Environment:**

- Secrets Store CSI Driver version: (use the image tag): v0.3.0
- Azure Key Vault provider version: (use the image tag): v0.2.0
- Kubernetes version: (use `kubectl version` and `kubectl get nodes -o wide`): v1.20.5
- Cluster type: (e.g. AKS, aks-engine, etc): AKS

**Comment by aramase** [comment]
Prior to v0.1.0, the provider wrote the files to the filesystem. In v0.1.0 we switched to the CSI driver writing the files instead of the provider. This is to standardize how the files are written to fs instead of each provider having their own implementation.

As part of this change, we also switched to using the atomic writer which is what Kubernetes uses today for all the volume types using symlink. This is an expected behavior and is now conformant with what Kubernetes recommends. For instance, mounting the Kubernetes secret as file will have the same behavior. Hope this helps.
